Core/Services/OfferService.cs

```diff
@@ -44,9 +44,9 @@ public async Task<OfferCreateDTO> CreateOfferAsync(OfferCreateDTO offerCreate, s
             offer.OfferCreatorId = userId;
             offer.CreationDate = DateTimeOffset.UtcNow;
             offer.IsClosed = false;
-            offer.CreatorRoleId = _roleRepository.GetRoleByName(offerCreate.Role);
-            offer.GoodCategoryId = _goodCategoryRepository
-                .GetGoodCategoryByName(offerCreate.GoodCategory);
+            offer.CreatorRoleId = await _roleRepository.GetRoleByNameAsync(offerCreate.Role);
```

Comment:
Please use [RoleManager ](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.rolemanager-1?view=aspnetcore-6.0) or explain why it should not be used